x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java

```diff
@@ -278,6 +283,28 @@ static ShutdownShardMigrationStatus shardMigrationStatus(
                 }
                 return hasShardCopyOnOtherNode == false;
             })
+            // If ILM is shinking the index this shard is part of, it'll look like it's unmovable, but we can just wait for ILM to finish
+            .filter(pair -> {
```

Comment:
Nit: just a though here: this method already looks very long and complex. Maybe it is possible to factor out this lambda into a private function with a nice name, so it's clearer and keeps the method complexity at bay?

Reply:
Good call, will do.